tutorials/arvo/arvo.md

```diff
@@ -0,0 +1,519 @@
+```
+
+```hoon
+=/  pit=vase  !>(..is)                                  ::
+```
+This `vase` is part of the state but does not get directly migrated when `+poke`
+is called. `!>(..is)` consists of the code in `arvo.hoon` written above this core contained in
+a `vase`. Thus this part of the state changes only when that code changes in an
+update.
+
+```hoon
+=/  vil=vile  (viol p.pit)                              ::  cached reflexives
+
+```
+This is a cache of specific types that are of fundamental importance to Arvo -
+namely `type`s, `duct`s, `path`s, and `vase`s. This is kept because it is
+unnecessarily wasteful to recompile these fundamental types on a regular basis.
+Again, this part of the state is never updated directly by `+poke`.
+
+```hoon
+=|  $:  lac=_&                                          ::  laconic bit
+        eny=@                                           ::  entropy
+        our=ship                                        ::  identity
+        bud=vase                                        ::  %zuse
+        vanes=(list [label=@tas =vane])                 ::  modules
+    ==
+```
+This is where the real state of the Arvo kernel is kept. `lac` detemines whether
+Arvo's output is verbose, which can be set using the `|verb` command in the
+dojo. `eny` is the current entropy. `our` is the ship, which is permanently
+frozen during the larval stage. `bud` is the standard library. Lastly, `vanes`
+is of course the list of vanes, which have their own internal states. As Hoon is
+homoiconic, code is data, so this is does secrely have everything the vanes are
+doing encoded within, including all user apps.
+
+As you can see, the state of Arvo itself is quite simple. Its primary role is that of
+a traffic cop, and most of interesting part of the state lies in `vanes`.
+
+
+## Vanes
+
+The Arvo kernel can do very little on its own. Its functionality is extended in a careful and controlled way with vanes, also known as kernel modules.
+
+As described above, we use Arvo proper to route and control the flow of `move`s.
+However, Arvo proper is rarely directly responsible for processing the event
+data that directly causes the desired outcome of a `move`. This event data is contained within a `card`. Instead, Arvo proper passes the `card` off to one of its vanes, which each present an interface to clients for a particular well-defined, stable, and general-purpose piece of functionality.
+
+As of this writing, we have nine vanes, which each provide the following services:
+
+- [Ames](@/docs/tutorials/arvo/ames.md): the name of both our network and the vane that communicates over it.
+- [Behn](@/docs/tutorials/arvo/behn.md): a simple timer.
+- [Clay](@/docs/tutorials/arvo/clay.md): our version-controlled, referentially- transparent, and global filesystem.
+- [Dill](@/docs/tutorials/arvo/dill.md): a terminal driver. Unix sends keyboard events to `%dill` from either the console or telnet, and `%dill` produces terminal output.
```

Comment:
I totally forgot we ever supported telnet.  Do we still support it?

Reply:
Hmmm I have you on my list as the person to talk to about Dill :P I'll poke around and see if that functionality still exists

Reply:
I started looking into the history of dill with git-timemachine and eventually ran into some weirdness. My skimming leads me to believe it does not currently support telnet. I'll just remove the reference to it here for now and look into it in more depth once I start updating the documentation on dill